crates/bevy_ecs/src/system/system.rs

```diff
@@ -55,6 +48,8 @@ pub trait System: Send + Sync + 'static {
     fn apply_buffers(&mut self, world: &mut World);
     /// Initialize the system.
     fn initialize(&mut self, _world: &mut World);
+    /// Update the system's cached archetypes.
```

Comment:
```suggestion
    /// Update the system's cached [`Archetypes`](crate::archetypes::Archetypes).
```

Reply:
If I can bikeshed a little, in my stageless branch, I've done this:

```rust
/// Updates the runtime access of the system to account for `archetype`.
fn add_archetype(&mut self, archetype: &Archetype);
/// Updates the runtime access of the system to account for any new [`Archetype`] in `world`.
fn add_new_archetypes(&mut self, world: &World);
```

Reply:
To counter-bikeshed: that first name is pretty unclear, and doesn't really match the comment ;)

I think the split and comments are good though.

Reply:
🤔, maybe something like `register_new_archetype` and `register_all_new_archetypes` then.

I just think `new_archetype` is vague and `update_archetypes` does not update any archetypes...

Reply:
Took the doc suggestion, still thinking about naming.